src/p4p/nt/__init__.py

```diff
@@ -199,40 +206,36 @@ def wrap(self, values, **kws):
                     V = cols[L]
                     if len(V) == 0:
                         del cols[L]
-
-                try:
-                    values = self.Value(self.type, {
-                        'labels': self.labels,
-                        'value': cols,
-                    })
-                except:
-                    _log.error("Failed to encode '%s' with %s", cols, self.labels)
-                    raise
+                update = {'labels': self.labels, 'value': cols}
+                if isinstance(values, list):
+                    try:
+                        values = self.Value(self.type, update)
+                    except:
+                        _log.error("Failed to encode '%s' with %s", cols, self.labels)
```

Comment:
I do not think it is desirable for a library to catch, log, and re-raise an exception in a situation where the method caller is able to do so directly (and may have reasons not to).